### PR TITLE
Support `action_plugin` in plugin_routing_schema

### PR DESCRIPTION
**Pull request by NilashishC**
##### SUMMARY

As per #77265 and [devel loader code](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/loader.py#L533-L534) as well, `action_plugin` should be a valid entry in `plugin_routing` schema.

Related PR: https://github.com/ansible-collections/cisco.nxos/pull/804

##### ISSUE TYPE

- Bugfix Pull Request (?)

### CODE REVIEW

File: test/lib/ansible_test/_util/controller/sanity/code-smell/runtime-metadata.py [hunk 1]
```diff
@@ -202,6 +202,7 @@ def validate_metadata_file(path, is_ansible, check_deprecation_dates=False):
             ('deprecation'): Any(deprecation_schema),
             ('tombstone'): Any(tombstoning_schema),
             ('redirect'): fqcr,
+            ("action_plugin"): fqcr,
```
Comment (s-hertel) [inline test/lib/ansible_test/_util/controller/sanity/code-smell/runtime-metadata.py]: This should only be valid for modules https://github.com/ansible/ansible/blob/devel/test/lib/ansible_test/_util/controller/sanity/code-smell/runtime-metadata.py#L236.

Comment (NilashishC) [reply test/lib/ansible_test/_util/controller/sanity/code-smell/runtime-metadata.py]: Updated.

**Review by s-hertel — APPROVED** [review]
Thanks for fixing this!

**Comment by webknjaz** [comment]
@bcoca @NilashishC @s-hertel any chance this is responsible for the coverage drop @ https://github.com/ansible/ansible/blob/aa40167/test/support/windows-integration/collections/ansible_collections/ansible/windows/plugins/plugin_utils/_reboot.py#L550-L553?

_(https://app.codecov.io/gh/ansible/ansible/commit/b01f1f207c9c46eda8f8e8ffab8bd4b99e03c754/indirect-changes)_

**Comment by bcoca** [comment]
@webknjaz  i would think #82526 is more likely culprit for that

**Comment by webknjaz** [comment]
@bcoca it shouldn't be, because the drop happened in between the last few commits. That merge was like a week ago.

**Comment by s-hertel** [comment]
I don't see how this would have changed the increased/decreased the coverage for any of the files in https://app.codecov.io/gh/ansible/ansible/commit/b01f1f207c9c46eda8f8e8ffab8bd4b99e03c754/indirect-changes :confused:

**Comment by webknjaz** [comment]
@bcoca @s-hertel this is puzzling because the only other commits in between the coverage change were https://github.com/ansible/ansible/pull/82569 and https://github.com/ansible/ansible/pull/82570...

**Comment by webknjaz** [comment]
@mattclay there's a small coverage drop that seems to be caused by this change, any idea on how to verify it? Is it hard to trigger a nightly-like CI run against the previous commit https://github.com/ansible/ansible/commit/853423c270d8e0ca983cbe77d79e738a228e6c8e?

**Comment by webknjaz** [comment]
Nevermind.. I figured out how to trigger it via the Azure Pipelines UI.

**Comment by webknjaz** [comment]
OTOH, that was quick. So a full CI run didn't happen.

**Comment by mattclay** [comment]
@webknjaz To trigger a full CI run with coverage, add the comments `ci_complete` and `ci_coverage` to the latest commit message on the PR.

**Comment by s-hertel** [comment]
> This commit b01f1f2 compared to [6935c8e](https://app.codecov.io/gh/ansible/ansible/commit/6935c8e303440addd3871ecf8e04bde61080b032)

@webknjaz The commit it's compared against shows an increase in coverage to that same file + section, but also seems unrelated https://app.codecov.io/gh/ansible/ansible/commit/6935c8e303440addd3871ecf8e04bde61080b032/indirect-changes#92e9c03abff1ec4c0a82181a533d5aa1-L547. In fact, looks like the coverage has been changing repeatedly for a while (https://app.codecov.io/gh/ansible/ansible/commit/4479c9df13aedec52f4971fb2a5a9c3c9fd92ba1/indirect-changes, https://app.codecov.io/gh/ansible/ansible/commit/8af920c8924b2fd9a0e4192c3c7e6085b687bfdc/indirect-changes). It seems like the coverage is just inconsistent.

**Comment by webknjaz** [comment]
@s-hertel my understanding is that we only get coverage uploads for the nightly runs. So we wouldn't see coverage for the intermediate commits if there's more than one merge per day.
That said, you might be onto something. I'll have to double check.

**Comment by s-hertel** [comment]
That reflects my understanding too. I don't see how that's related to my comment.

**Pull request merged**